### ISSUE TITLE
Missing documentation on how to mock ActivatedRoute

### ISSUE THREAD

**Issue by DaSchTour** [angular/angular#59143]
### Describe the problem that you experienced

This comment: https://github.com/angular/angular/issues/54438#issuecomment-1945010134 triggered me to create this issue, as I was fighting with mocking ActivatedRoute and have a big class that creates some kind of more or less stable mock for ActivatedRoute.

This statement throw me of
> We see this all the time with developers mocking ActivatedRoute instead of using the router testing utilities.

I know that there are router testing utilities but I have now idea how to use them to create an ActivatedRoute that I can insert into a functional guard as parameter when testing it.

The documentation lacks any explanation or example of how testing guards or resolvers is envisioned by the angular team, while there seams to be a clear idea of how we developers should to it but it's not communicated or documented.

### Describe what you were looking for in the documentation

In-depth Guides →  Routing →  Testing
In-depth Guides →  Testing →  Routing

### Describe the actions that led you to experience the problem

Reading the mentioned comment: https://github.com/angular/angular/issues/54438#issuecomment-1945010134

### Describe what you want to experience that would fix the problem

Examples for testing guards and resolvers. And how to create an ActivatedRoute or even just an ActivatedRouteSnapshot.

### If this problem caused an exception or error, please paste it here

```true

```

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true

```

### Provide any additional information here in as much as detail as you can

```true
Maybe also exposing some of the currently private API for creating RouterState might help to be able to create better experience in testing router related stuff. Or maybe not exposing them directly but make them somehow usable through router testing utils.
```

**Comment by atscott** [comment]
Mocking `ActivatedRoute` is not recommended. Tests should trigger real navigations to get a real instance of `ActivatedRoute`, and this is made easier with the [`RouterTestingHarness`](https://angular.dev/api/router/testing/RouterTestingHarness). A longer article about this decision and recommendation can be found on the [blog](https://blog.angular.dev/write-better-tests-without-router-mocks-stubs-bf5fc95c1c57).

**Comment by DaSchTour** [comment]
Yeah that's what I've done now in many cases. Feels somehow strange, that for my one line functional guards I have 20 times more code to just setup the tests.

But I think the point of my issue is still valid. There is **no** guidance about testing in the documentation. I feel like this area gets to few attention.

**Comment by atscott** [comment]
> There is no guidance about testing in the documentation

New router documentation is currently being drafted. There will be something available in the coming months.
